### DISCUSSION TITLE
Screen resolution using "getUnstableScreenResolution" with somewhat arbitrary sorting

### DISCUSSION THREAD

**Discussion by oofpez** [fingerprintjs/fingerprintjs#1057] [General]
https://github.com/fingerprintjs/fingerprintjs/blob/7e4fa4248641212b734d77a29a3a7bfe1a94bfc9/src/sources/screen_resolution.ts#L33

The screen resolutions of width and height are arbitrarily sorted using
.sort().reverse()

1. This reduces the quality of the data since we lose context of which is height and which is width.

2. On top of this though, sort() on integers casts them to strings before sorting them if a compareFn is not implemented, so it doesn't sort accurately.

3. using a compareFn would also allow specifying the sort order, so the .reverse() would not be necessary.

Suggestion: 
use a simple inline compareFn like

.sort((a, b) => b - a)

or just remove sorting entirely.

**Comment by Valve** [comment]
Hi @oofpez 
Thanks for your suggestion.

Can you please clarify why the suggested sorting is better for the screen resolution fingerprinting component?
It looks like a stylistic suggestion that doesn't help the fingerprinting process.

**Reply by oofpez** [reply]
I'm not sure why they are sorted in the first place. It would be ideal to have width and height consistent in order, so if somebody wants to use this information later on they don't have to guess which is which.


My suggestion on improving the sorting has to do with the default .sort() implementation in javascript, which sorts the elements as strings in alphabetical order.

If you test the following:

// Create an Array
const points = [40, 100, 1, 5, 25, 10];

// Sort the Array
points.sort(function(a, b){return b-a});
console.log(points)
points.sort().reverse();
console.log(points)

You should see the following results

[100, 40, 25, 10, 5, 1]
[5, 40, 25, 100, 10, 1]

Clearly the .sort().reverse() implementation comparing numbers alphabetically is wrong

**Comment by Finesse** [comment]
Hi @oofpez. The screen dimensions are sorted to stabilize the entropy source. When you rotate a mobile device, the dimensions swap, which changes the whole fingerprint. In fact, the sorting may be disabled on desktop/laptop devices because they almost never rotate (as far as I know). If you want, you may create a PR.

> On top of this though, sort() on integers casts them to strings before sorting them if a compareFn is not implemented, so it doesn't sort accurately.

The actual order doesn't matter. We only need the dimensions not to swap.
